src/evo/netinfo.cpp

```diff
@@ -24,6 +25,130 @@ const CChainParams& MainParams()
 }
 } // anonymous namespace
 
+bool NetInfoEntry::operator==(const NetInfoEntry& rhs) const
+{
+    if (m_type != rhs.m_type) return false;
+    return std::visit(
+        [](auto&& lhs, auto&& rhs) -> bool {
+            if constexpr (std::is_same_v<decltype(lhs), decltype(rhs)>) {
+                return lhs == rhs;
+            }
+            return false;
+        },
+        m_data, rhs.m_data);
+}
+
+bool NetInfoEntry::operator<(const NetInfoEntry& rhs) const
+{
+    if (m_type != rhs.m_type) return m_type < rhs.m_type;
+    return std::visit(
+        [](auto&& lhs, auto&& rhs) -> bool {
+            using T1 = std::decay_t<decltype(lhs)>;
+            using T2 = std::decay_t<decltype(rhs)>;
+            if constexpr (std::is_same_v<T1, T2>) {
```

Comment:
a bit too many conditions and returns.... maybe simplify a bit?

```
            if constexpr (std::is_same_v<T1, T2>) {
                // Both the same type, compare as usual
                return lhs < rhs;
            }
            // If lhs is monostate, it less than rhs; otherwise rhs is bigger.
            return std::is_same_v<T1, std::monostate>;
```